### ISSUE TITLE
How to handle value shift for sleep data when sleepting and waking up on the same day?

### ISSUE THREAD

**Issue by wenlzhang** [pyrochlore/obsidian-tracker#64]
Data for Day 1: `Sleep:: 02:00 am/11:00 am`; Data for Day 2: `Sleep:: 10:00 pm/06:00 am`.

The query:

```tracker
searchType: dvField
searchTarget: Sleep[0], Sleep[1]
startDate: 2021-08-30
endDate: 2021-09-05
valueShift: -24:00, 00:00
datasetName: Sleep, WakeUp
line:
    title: Sleep
    yAxisLabel: "Time (24h)"
    lineColor: green, red
	showPoint: true
	showLegend: true
	fillGap: true
```

As can be seen from the following figures:

- The Sleep and WakeUp time would look a little weird.
- The y-axis values are overlapped. Is there any workaround to fix this?
 
Also, when defining the data with `Sleep:: 10:00 pm/06:00 am`, it works; but it does not work with `Sleep:: 10:00 pm, 06:00 am`. Maybe both separators should be supported for dataview fields?

The figure:
<img width="391" alt="Screenshot 2021-09-01 at 18 43 05" src="https://user-images.githubusercontent.com/44930227/131710567-338fb4c7-0f24-43c9-aebc-7a009d09c116.png">

**Comment by pyrochlore** [comment]
The valueShift is a constant value for each collected data point of a target, but that would be a problem for your data. 
In your case, 02:00 am shouldn't be shifted.
I will figure out a way to solve this problem.

Other questions you mentioned.

- The Sleep and WakeUp time would look a little weird.

Do you mean the text arrangement in the legend? It should be fixed.

- The y-axis values are overlapped. Is there any workaround to fix this?

This should be fixed too.

- Also, when defining the data with Sleep:: 10:00 pm/06:00 am, it works; but it does not work with Sleep:: 10:00 pm, 06:00 am. Maybe both separators should be supported for dataview fields?

You can use parameter `separator` like this
```
...
separator: comma
line:
```

**Comment by wenlzhang** [comment]
> Do you mean the text arrangement in the legend? It should be fixed.

I meant the value shift thing. And it would be great that the text arrangement in the legend would also be fixed.

> You can use parameter `separator` like this

I thought comma was the default separator. Sorry about this.

**Comment by pyrochlore** [comment]
Hi @wenlzhang,

The corresponding enhancement and fix were just released in version 1.10.4. Including

1. Fixed text arrangement in the legend
2. Add a new parameter `shiftOnlyValueLargerThan` to determine when to do the shift.

For example, if we set a value of "12:00" to `shiftOnlyValueLargerThan`, 
the collected value '10:00 pm' will be shifted and '02:00 am' will not.
Here is a simple demo

![image](https://user-images.githubusercontent.com/3617615/132944533-a196ce25-86ce-4f48-8edc-f4d2b9cf70ff.png)
